### ISSUE TITLE
Automagic backup of last modlist marked for installation for when poo->fan()

### ISSUE THREAD

**Issue by Dazpoet** [KSP-CKAN/CKAN#1395]
I think we all know that CKAN seldom crashes while we're installing a single mod which we'd just like to test out. Rather it saves the crash for after we painstakingly selected and sorted a long fine list of mods which we want.

As I understand it the installation of mods through ckan, in layman terms, works something like:

1) select mods
2) run selections based on TMP/virtual packages
3) run installation
4) end transactions

I think it should be

1) select mods
2) run selections based on TMP/virtual packages
3) create a metapackage over all the mods in the current list we want to install
4) run installation
5) end transactions
6) remove the metapackage created in 3)

This way if CKAN crashes before 6) for any reason the user can click a nice little button in the GUI saying "reload last installed modselection" or something to the same effect and automagically have the previously selected mods marked again without the tedium of re-clicking them.

It might be advisable to even have the metapackage created before resolving TMPs if those are still an issue with metapackages?

**Comment by Postremus** [comment]
metapackages still throw tmp's. see #1274

**Comment by Dazpoet** [comment]
So I guess a solution to this would then depend on https://github.com/KSP-CKAN/CKAN/pull/1108 ?

**Comment by Postremus** [comment]
it depends on either #1108 or #1337.

**Comment by pjf** [comment]
This seems like a great idea. :)

**Comment by Noiredd** [comment]
Just bumping this up. Last time I used CKAN was 2 years ago and it annoyed the hell out of me (while installing RO for example), shame to see this still being an issue now.

**Comment by politas** [comment]
Would we have to change the handling of metapackages, so that it basically sets all the checkboxes, and lets the user then review and hit the commit button?

**Comment by HebaruSan** [comment]
We save a .ckan file after every install as of #2239.

![image](https://user-images.githubusercontent.com/1559108/82130031-56c2a080-978d-11ea-8e71-2d153ffdc938.png)

There's no user friendly automatic revert capability, but it comes up pretty rarely.
